### PR TITLE
(Fix) Add address to whitelist in manage screen

### LINKED ISSUE #596: (Bug) Modifiable crowdsale: incorrect behavior if user tries to add existing whitelisted address
### Steps to reproduce:
1. Create modifiable crowdsale with whitelisted addresses
2. Open manage 
3. Add in the whitelist existed whitelisted address, fill out min/max value
4. Click button "+"
5. Click button "Save"

Expected result: should be error message like "address already exist ..."
                           OR user should be allowed to modify existed addresses(Feature)
Actual result: button "Save' is enabled, user can proceed, message "Congrats. You've successfully updated the crowdsale "

If you are reporting a problem with Token Wizard, please include the following information:

### Which network did you use? (Mainnet, Kovan, Rinkeby, etc.)

*Rinkeby*

### If you were able to create it, what is the URL of your crowdsale?

*https://wizard.poa.network/invest?addr=0x011C0608e9858f22564C31199438f9a732B6f157&networkID=4*

### Do you have screenshots showing the problem?

incorrbehavior

https://drive.google.com/open?id=1iAliRoG85jqjbjE1Ws_B-Q33yJsUCUSy

### Do you see errors in the dev console? If yes, please include a screenshot

To open the dev console in Google Chrome, press F12, or go to `View -> Developer -> Developer Tools`, and open the Console tab

If you see errors, please right click on them and "Save as..". Zip saved file and attach it to the Issue.

*No*

---

### PR DESCRIPTION
**Pull request by fernandomg**
Closes #596

In this PR `whitelistElements` collection was removed and only uses `whitelist`.

Style was applied for the whitelist elements in the manage screen, so the user can recognize what addresses are being modified, added or untouched.

image

Things to fix that I found: 
1. The way how duplicated addresses are being checked. As the same address can have upper and lower cases, this may lead errors.
2. `min` and `max` values are not being checked, so a user can enter a `min` value greater than `max`. The tx will fail, but it won't be clear why.

### CODE REVIEW

File: src/stores/TierStore.js [hunk 1]
```diff
@@ -168,33 +168,60 @@ class TierStore {
...
+    whitelist.push(newItem)
+    this.sortWhitelist(crowdsaleNum)
+  }
+
+  @action sortWhitelist = (crowdsaleNum) => {
+    this.tiers[crowdsaleNum].whitelist = this.tiers[crowdsaleNum].whitelist.sort((prev, curr) => {
+      const currentAddress = curr.addr.toLowerCase()
+      const previousAddress = prev.addr.toLowerCase()
+
+      return currentAddress > previousAddress ? -1 : currentAddress === previousAddress ? curr.stored ? 1 : -1 : 1
```
Comment (fvictorio) [inline src/stores/TierStore.js]: This needs tests.

File: src/stores/TierStore.js [hunk 2]
```diff
@@ -168,33 +168,60 @@ class TierStore {
...
+    })
   }
 
   @action removeWhitelistItem = (whitelistNum, crowdsaleNum) => {
-    let whitelist = this.tiers[crowdsaleNum].whitelist.slice()
-    whitelist[whitelistNum].deleted = true
-    this.setTierProperty(whitelist, 'whitelist', crowdsaleNum)
+    const removedItem = this.tiers[crowdsaleNum].whitelist.splice(whitelistNum, 1)[0]
```
Comment (fvictorio) [inline src/stores/TierStore.js]: This needs tests too.

File: src/components/Common/WhitelistInputBlock.js [hunk 3]
```diff
@@ -152,15 +152,12 @@ export class WhitelistInputBlock extends React.Component {
             />
           </div>
         </div>
-        {whitelistElements && whitelistElements.map(e =>
+        {whitelist.length > 0 && whitelist.map((item, index) =>
```
[the remark this answers is not in the document]
Comment (fernandomg) [reply src/components/Common/WhitelistInputBlock.js]: fixed

File: src/stores/TierStore.js (hunk 1 above)
Comment (fernandomg) [reply src/stores/TierStore.js]: done

File: src/stores/TierStore.js (hunk 2 above)
Comment (fernandomg) [reply src/stores/TierStore.js]: done, and added test for the other modified/created methods

**Comment by fernandomg** [comment]
@vbaranov conflict resolution done

**Comment by fvictorio** [comment]
@vbaranov Can we merge this?

**Comment by vbaranov** [comment]
@fvictorio formally yes, but if you could give me some hours, I'll appreciate it. I plan to test it too soon.

**Comment by fvictorio** [comment]
Sure, no problem. I just want to merge as much PRs as possible so that I can start working on #536 with less risk.

**Pull request merged**